### PR TITLE
sys/phydat: dump function can handle fixed floating points

### PR DESCRIPTION
**Pull request by haukepetersen**
This PR should beautify the output of values for some cases, instead of `219E-1°C` it will now print `21.0°C'.

@kaspar: Mind to take a look at the new 'fmt_s16_ffp()` function?

~~TODO:~~
~~- doxygen is basically untouched~~

### CODE REVIEW

File: sys/fmt/fmt.c [hunk 1]
```diff
@@ -118,6 +123,25 @@ size_t fmt_s32_dec(char *out, int32_t val)
     return fmt_u32_dec(out, val) + negative;
 }
 
+size_t fmt_s16_dec(char *out, int16_t val)
```
Comment (cgundogan) [inline sys/fmt/fmt.c]: is this function really necessary if it's just a wrapper around `fmt_s32_dec`? The user could call `fmt_s32_dec` directly.
If you insist on keeping `fmt_s16_dec` we could at least inline it?

Comment (haukepetersen) [inline sys/fmt/fmt.c]: AFAIK the compiler will inline it automatically anyway, same goes for `fmt_u16_dec`...

Comment (kaspar030) [inline sys/fmt/fmt.c]: The compiler can only inline functions of the same compilation unit.

File: sys/phydat/phydat_str.c [hunk 2]
```diff
@@ -32,10 +32,27 @@ void phydat_dump(phydat_t *data, uint8_t dim)
...
+
+        if (scale_str) {
+            printf("%i%c", (int)data->val[i], scale_str);
+        }
+        else if (data->scale == 0) {
+            printf("%i", (int)data->val[i]);
+        }
+        else if ((data->scale > -6) && (data->scale < 0)) {
+            char num[8];
+            size_t len = fmt_s16_ffp(num, data->val[i], data->scale * -1);
```
Comment (cgundogan) [inline sys/phydat/phydat_str.c]: does it make sense to convert all printfs/puts to `fmt_` functions? This way `phydat` wouldn't rely on `stdio` and `fmt`

Comment (haukepetersen) [inline sys/phydat/phydat_str.c]: could be an option, but this is unrelated to this PR. In this particular case above I am using fmt, as stdio does not provide the feature I need... (or does it?)

File: sys/fmt/fmt.c (hunk 1 above)
Comment (haukepetersen) [inline sys/fmt/fmt.c]: True... 

Back to the question: I don't care if it is inlined or not, I just copied what was there already... I would go with this now, feel free to PR a fix with inlining some of the functions.

Comment (cgundogan) [inline sys/fmt/fmt.c]: +1
Am 05.02.2016 17:18 schrieb "Hauke Petersen" <notifications@github.com>:

**Comment by haukepetersen** [comment]
@kaspar030, @cgundogan: are you good with this PR? Then I will make it ready for merge...

File: sys/include/fmt.h [hunk 3]
```diff
@@ -127,6 +129,10 @@ size_t fmt_u16_dec(char *out, uint16_t val);
  */
 size_t fmt_s32_dec(char *out, int32_t val);
 
+size_t fmt_s16_dec(char *out, int16_t val);
```
Comment (kaspar030) [inline sys/include/fmt.h]: missing dox?

**Comment by cgundogan** [comment]
@haukepetersen ack for the general idea, please retouch this PR to make it shiny before merging .. docs etc.

**Comment by haukepetersen** [comment]
Alright, now it should be shiny: added doxygen, added unittests and fixed many issues that I found when running the unittests...

**Comment by cgundogan** [comment]
Murdock agrees - GO
